src/lib.rs

```diff
@@ -542,6 +572,8 @@ pub trait PolynomialCommitment<G: AffineCurve>: Sized {
         let x_point: G::ScalarField = fs_rng.squeeze_128_bits_challenge();
 
         // Assert x_point != x_1, ..., x_m
+        // This is needed as we use a slightly optimized LC, which costs one 
+        // scalar multiplication. 
         if eval_points.into_iter().any(|eval_point| eval_point == &x_point) {
```

Comment:
Is it really needed? The probability that x in {x_1,..., x_m} is negligible...

Reply:
In general - yes, but any way it is not zero and in audit we've got this as a warning during the audit